server/sonar-server/src/main/java/org/sonar/server/qualityprofile/ws/DeleteAction.java

```diff
@@ -87,13 +87,13 @@ public void handle(Request request, Response response) {
 
   private static void ensureNoneIsMarkedAsDefault(QualityProfileDto profile, List<QualityProfileDto> descendants) {
     if (profile.isDefault()) {
-      throw new IllegalStateException("Profile is marked as 'default' and cannot be deleted");
+      throw new IllegalArgumentException("Profile is marked as 'default' and cannot be deleted");
     }
     descendants.stream()
       .filter(QualityProfileDto::isDefault)
       .findFirst()
       .ifPresent(p -> {
-        throw new IllegalStateException("Profile cannot be deleted because its descendant named [" + p.getName() + "] is marked as 'default'");
+        throw new IllegalArgumentException("Profile cannot be deleted because its descendant named [" + p.getName() + "] is marked as 'default'");
```

Comment:
The message is changed. I do not want to do the refactoring of `checkArgument` right now, for to save a bit of time for my other tasks.